lib/gitsh/environment.rb

```diff
@@ -39,7 +40,7 @@ def fetch(key, force_default_git_command = false, &block)
         end
       end
     rescue KeyError
-      raise KeyError, "Variable '#{key}' is not set"
+      raise Gitsh::Error, "Variable '#{key}' is not set"
```

Comment:
It often proves useful to know what kind of exception happened without string parsing. This is the first exception class, so it doesn't matter yet, but how about starting off with a `Gitsh::KeyError` subclass here?